### PR TITLE
[KOGITO-3094] - Enable persistence selection in Data Index controller

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-3094

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
@radtriste for bdd tests, probably needs to be added after operator is able to handle the mongodb infra as well, right?

File: pkg/controller/kogitodataindex/kogitodataindex_controller.go [hunk 1]
```diff
@@ -175,6 +174,26 @@ func (r *ReconcileKogitoDataIndex) onDeploymentCreate(cli *client.Client, deploy
...
+				infrastructure.PersistenceProviders,
+				persistenceProvider)
+		}
+	} else {
+		log.Infof("the % property was not found, defaulting persistence provider to %s",
+			infrastructure.DataIndexPersistenceTypeProp, persistenceProvider)
+	}
+
+	log.Infof("injecting %s with the persistence provider %s.", infrastructure.DataIndexPersistenceEnv, persistenceProvider)
+	framework.SetEnvVar(infrastructure.DataIndexPersistenceEnv, persistenceProvider, &deployment.Spec.Template.Spec.Containers[0])
```
Comment (ricardozanini) [inline pkg/controller/kogitodataindex/kogitodataindex_controller.go:196]: Why do we need this env var? :thinking:

**Review by ricardozanini — COMMENTED** [review]
I'm not sure if we need this, @spolti . What's the behavior of the image having the env var set?

File: pkg/controller/kogitodataindex/kogitodataindex_controller.go (hunk 1 above)
Comment (spolti) [reply pkg/controller/kogitodataindex/kogitodataindex_controller.go:196]: i don't know what happened here, for some reason it reused my last changes. will update it tomorrow. just ignore it.

Comment (spolti) [reply pkg/controller/kogitodataindex/kogitodataindex_controller.go:196]: oh crap, I messed up with the branch names.
moved to https://github.com/kiegroup/kogito-cloud-operator/pull/621

**Pull request closed without merging**